confirm.c

```diff
@@ -763,16 +765,18 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
-confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
-		confirm_query_cb cb)
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
 {
 	struct confirm_query *query;
-
-	query = malloc(sizeof(struct confirm_query));
+	query = malloc(sizeof(*query));
 	if(!query) logea(__FILE__, __LINE__, NULL);
 
+	if(flowid > CONFIRM_MAX_FLOWID) {
+		logd(LOG_WARN, "%s,%d: flowid > 127!\n", __FILE__, __LINE__);
```

Comment:
New issue here. https://github.com/cunha/netlib/issues/17

If you have any ideas, please put them in the issue from now on (this pull request review page is a bit confusing).